build.assets/macos/tshdev/README.md

```diff
@@ -79,3 +81,104 @@ through the steps here unless you are trying to recreate parts of the skeleton
 
     For example, click "build" and copy the app using "Product -> Show Build
     Folder in Finder".
+
+## Working with launch daemons
```

Comment:
Will every execution of a signed tsh register the login item?

What do I do if I don't want that to happen?

Reply:
No, the login item is registered only when `tsh vnet` is ran, because that's the only code path that [registers the daemon](https://developer.apple.com/documentation/servicemanagement/smappservice/register()?language=objc). Any other execution of a signed tsh is not going to register the login item.

Something to remember in the future is that enabling one login item from a bundle enables all of them. I saw that mentioned in one of those Apple dev forum threads, but I forgot to capture a link.

I suppose this also addresses the second question, doesn't it?

I guess one improvement I could make is to skip the call to register if the service is already enabled.

Reply:
That clears some of it up, thanks.

I think it's worth having a way to skip the registration for dev builds, even if it's just for `tsh vnet`.

> Something to remember in the future is that enabling one login item from a bundle enables all of them. I saw that mentioned in one of those Apple dev forum threads, but I forgot to capture a link.

That's gonna be a surprise if it happens, for sure. Thanks for flagging it now.

> I guess one improvement I could make is to skip the call to register if the service is already enabled.

If attempting to register is cheap/fast then I don't think it's much of a problem, assuming I understand this correctly.

Reply:
> I think it's worth having a way to skip the registration for dev builds, even if it's just for tsh vnet.

Nevermind this part here, https://github.com/gravitational/teleport/pull/43746#discussion_r1666079146 has a better thread on this.